beta/src/content/apis/react/Component.md

```diff
@@ -1,28 +1,725 @@
+
+```js
+import { Component } from 'react';
+
+class Greeting extends Component {
+  render() {
+    return <h1>Hello, {this.props.name}!</h1>;
+  }
+}
+```
+
+Your component's props will be available as [`this.props`.](#props)
+
+<Sandpack>
+
+```js
+import { Component } from 'react';
+
+class Greeting extends Component {
+  render() {
+    return <h1>Hello, {this.props.name}!</h1>;
+  }
+}
+
+export default function App() {
+  return (
+    <>
+      <Greeting name="Sara" />
+      <Greeting name="Cahal" />
+      <Greeting name="Edite" />
+    </>
+  );
+}
+```
+
+</Sandpack>
+
+A class component's `render` method is similar to a regular [function component.](/learn/your-first-component#defining-a-component) It also needs to return JSX and should be [written as a pure function.](/learn/keeping-components-pure) However, you can't call any Hooks (such as [`useState`](/apis/react/useState)) inside a class.
+
+<Pitfall>
+
+We don't recommend to define newly written components as classes. [See the alternatives.](#migrating-a-simple-component-from-a-class-to-a-function)
+
+</Pitfall>
+
+---
+
+### Adding state to a class component {/*adding-state-to-a-class-component*/}
+
+To add [state](/learn/state-a-components-memory) to a class, assign an object to a property called [`state`](#state). To update state, call [`this.setState`](#setstate).
+
+<Sandpack>
+
+```js
+import { Component } from 'react';
+
+export default class Counter extends Component {
+  state = {
+    name: 'Taylor',
+    age: 42,
+  };
+
+  handleNameChange = (e) => {
+    this.setState({
+      name: e.target.value
+    });
+  }
+
+  handleAgeChange = () => {
+    this.setState({
+      age: this.state.age + 1 
+    })
+  };
+
+  render() {
+    return (
+      <>
+        <input
+          value={this.state.name}
+          onChange={this.handleNameChange}
+        />
+        <button onClick={this.handleAgeChange}>
+          Increment age
+        </button>
+        <p>Hello, {this.state.name}. You are {this.state.age}.</p>
+      </>
+    );
+  }
+}
+```
+
+```css
+button { display: block; margin-top: 10px; }
+```
+
+</Sandpack> 
+
+<Pitfall>
+
+We don't recommend to define newly written components as classes. [See the alternatives.](#migrating-a-component-with-state-from-a-class-to-a-function)
+
+</Pitfall>
+
+---
+
+### Adding lifecycle methods to a class component {/*adding-lifecycle-methods-to-a-class-component*/}
+
+There are a few special methods you can define on your class.
+
+If you define the [`componentDidMount`](#componentdidmount) method, React will call it when your component is first added *(mounted)* to the screen. React will call [`componentDidUpdate`](#componentdidupdate) after your component re-renders due to changed props or state. React will call [`componentWillUnmount`](#componentwillunmount) after your component has been removed *(unmounted)* from the screen.
```

Comment:
> React will call componentWillUnmount after your component has been removed

Shouldn't this be:
React will call componentWillUnmount before your component has been removed
?

Reply:
Hm I guess you're right.